agent/api/errors.go

```diff
@@ -25,6 +25,11 @@ func IsInstanceTypeChangedError(err awserr.Error) bool {
 	return strings.Contains(err.Message(), INSTANCE_TYPE_CHANGED_ERROR_MESSAGE)
 }
 
+func IsInvalidAttributeError(err awserr.Error) bool {
```

Comment:
Yes... This will need API changes. We currently don't have a way to distinguish between attribute related errors, which we want to treat as fatal, and the broader class of ClientException errors that should trigger retries.